src/Symfony/Component/Form/Extension/Csrf/EventListener/CsrfValidationListener.php

```diff
@@ -66,7 +66,7 @@ public function preSubmit(FormEvent $event)
                     $errorMessage = $this->translator->trans($errorMessage, array(), $this->translationDomain);
                 }
 
-                $form->addError(new FormError($errorMessage));
+                $form->addError(new FormError($errorMessage, null, array(), null, new CsrfValidationError()));
```

Comment:
I think it might be good to pass the `CsrfToken` instance as well as the submitted token to `CsrfValidationError` as that would ease debugging IMO.

Reply:
Hm... I have my doubts about this... CsrfToken does not always exists. Check the last failing build. It will need some refactoring there making the code I little bit more complex. Do you want me to proceed with the refactoring?